src/parser/EntitySearchQuery.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Felix Meisen (fesemeisen@outlook.de)
+
+#include "parser/EntitySearchQuery.h"
+
+#include "parser/MagicServiceIriConstants.h"
+
+namespace parsedQuery {
+
+// ____________________________________________________________________________
+void EntitySearchQuery::addParameter(const SparqlTriple& triple) {
+  auto sipmleTriple = triple.getSimple();
+  TripleComponent predicate = sipmleTriple.p_;
+  TripleComponent object = sipmleTriple.o_;
```

Comment:
I won't look at this class in detail, as I first have a general qustion:
Shoult thi bea separate SERVICE, or is the `contains-entity` stuff not part of the Fulltext search query?
So you have a single query (syntax imagined)
```
SERVICE textSearch: {
[] <textVar> ?text ;
<wordMatch> [<word> "walk"; <score> ?scoreWalk];
<entityMention> [<entity> <Flixtastic>; <score> ?scoreFelix] .
}
```

Of course then the `configToTextIndexScans` function always returns a vector of plans
(n WordScans, and m EntityScans), but we have a single class that manages the text index. Note: Not every word search has an entity search, but every entity search has a word search.
Maybe think about this, and we can also discuss this 1-1 next week some time.